pkg/util/limitrange/limitrange.go

```diff
@@ -81,20 +81,30 @@ func (s Summary) validatePodSpecContainers(containers []corev1.Container, path *
 }
 
 // TotalRequests computes the total resource requests of a pod.
-// total = sum(max(sum(.containers[].requests), initContainers[].requests), overhead)
+// total = Sum(Max( Max(nonSidecarInitContainers) + Sum(SidecarContainers), Sum(SidecarContainers) + Sum(Containers) ), overhead)
 func TotalRequests(ps *corev1.PodSpec) corev1.ResourceList {
-	total := corev1.ResourceList{}
-
 	// add the resource from the main containers
+	sumContainers := corev1.ResourceList{}
 	for i := range ps.Containers {
-		total = resource.MergeResourceListKeepSum(total, ps.Containers[i].Resources.Requests)
+		sumContainers = resource.MergeResourceListKeepSum(sumContainers, ps.Containers[i].Resources.Requests)
 	}
 
-	// take into account the maximum of any init containers
+	// take into account the maximum of any non sidecar init containers and add the sidecar containers
+	maxNonSidecarInitContainers := corev1.ResourceList{}
+	sumSidecarContainers := corev1.ResourceList{}
```

Comment:
nit: factor this logic out into a function for sidecar and init containers

Reply:
unless you will implement the second formula in the kep; which seems to require calculation together

Reply:
Do you mean that there should be a function to compute the total for init containers that returns those two values? Or one function for sidecar and another one for nonSidecar containers?

Reply:
I'd lean towards two functions, but this is up to you :)